common/static/js/src/lang_edx.js

```diff
@@ -18,32 +15,49 @@ var edx = edx || {},
              */
             listenForLanguagePreferenceChange: function() {
                 settings_language_selector.change(function(event) {
-                    var language = this.value;
+                    var language = this.value,
+                        url = $('.url-endpoint').val(),
+                        is_user_authenticated =  JSON.parse($('.url-endpoint').data('user-is-authenticated'));
                     event.preventDefault();
-                    $.ajax({
-                        type: 'PATCH',
-                        data: JSON.stringify({'pref-lang': language}) ,
-                        url: preference_api_url,
-                        dataType: 'json',
-                        contentType: "application/merge-patch+json",
-                        beforeSend: function (xhr) {
-                            xhr.setRequestHeader("X-CSRFToken", $('#csrf_token').val());
+                    self.submitAjaxRequest(language, url, function() {
+                        if (is_user_authenticated) {
+                            // User language preference has been set successfully
+                            // Now submit the form in success callback.
+                            $("#language-settings-form").submit();
```

Comment:
Nitpick: Please be consistent with quotes. The rest of the file uses single quotes so please update these double quotes to be single too.